scikeras/utils/_transformers.py

```diff
@@ -0,0 +1,262 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Union
+
+import numpy as np
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer, OneHotEncoder, OrdinalEncoder
+from tensorflow.python.keras.losses import is_categorical_crossentropy
+
+
+class Ensure2DTransformer(TransformerMixin, BaseEstimator):
+    """Transforms from 1D -> 2D and back.
+    """
+
+    def fit(self, X: np.ndarray) -> "Ensure2DTransformer":
+        self.should_transform_ = X.ndim == 1
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        if self.should_transform_:
+            if not (X.ndim == 1 or (np.prod(X.shape) == X.size)):
+                raise ValueError(
+                    "Expected a 1D array for `X` "
+                    "or for X to have one meaningful dimension (e.g, "
+                    "`X.shape == (1, 400, 1)`). Instead, got a"
+                    f" {X.ndim}D array instead with shape={X.shape}."
+                )
+            X = X.reshape(-1, 1)
+        return X
+
+    def inverse_transform(self, X: np.ndarray) -> np.ndarray:
+        if self.should_transform_:
+            if X.ndim != 2:
+                raise ValueError(
+                    f"Expected a 2D array of shape (n_samples, 1)"
+                    " for `X`, but got a"
+                    f" {X.ndim}D array instead."
+                )
+            if X.shape[1] != 1:
+                raise ValueError(
+                    f"Expected `X.shape[1] == 1`"
+                    f" but got `X.shape[1] == {X.shape}` instead."
+                )
+            X = np.squeeze(X, axis=1)
+        return X
+
+
+class BaseScikerasDataTransformer(TransformerMixin, BaseEstimator, ABC):
+    @abstractmethod
+    def fit(self, X: np.ndarray) -> "BaseScikerasDataTransformer":
+        pass
+
+    @abstractmethod
+    def transform(
+        self, X: np.ndarray
+    ) -> Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]:
+        pass
+
+    @abstractmethod
+    def inverse_transform(self, X: Union[List[np.ndarray], np.ndarray]) -> np.ndarray:
+        pass
+
+    @abstractmethod
+    def get_meta_params(self) -> Dict[str, Any]:
+        pass
+
+
+class BaseKerasClassifierTargetTransformer(BaseScikerasDataTransformer):
+    """Base class for KerasClassifier target transformers.
+    """
+
+    @abstractmethod
+    def inverse_transform(
+        self, X: Union[List[np.ndarray], np.ndarray], return_proba: bool = False
+    ) -> np.ndarray:
+        pass
```

Comment:
All of these "Base" transformers are here to make the point that `KerasClassifier` expects a different signature for `inverse_transform`, but I think maybe it's overkill to have a base for each wrapper?